GliaWidgets/Sources/CallVisualizer/Coordinator/CallVisualizer.Coordinator.swift

```diff
@@ -153,16 +50,126 @@ extension CallVisualizer {
 
             return icon
         }()
-        private var visitorCodeCoordinator: VisitorCodeCoordinator?
-        private var screenSharingCoordinator: ScreenSharingCoordinator?
-        private var videoCallCoordinator: VideoCallCoordinator?
+    }
+}
+
+extension CallVisualizer.Coordinator {
+    func resume() {
+        switch state {
+        case .initial:
+            return
+        case .screenSharing:
+            showEndScreenSharingViewController()
+        case .videoCall:
+            resumeVideoCallViewController()
+        }
+        environment.eventHandler(.maximized)
+    }
+
+    func showVisitorCodeViewController(by presentation: CallVisualizer.Presentation) {
+        let coordinator = CallVisualizer.VisitorCodeCoordinator(
+            theme: environment.viewFactory.theme,
+            environment: .create(with: environment),
+            presentation: presentation
+        )
+
+        coordinator.delegate = { [weak self] event in
+            switch event {
+            case .closeTap:
+                self?.visitorCodeCoordinator = nil
+                self?.environment.log.prefixed(Self.self).info("Dismiss Visitor Code Dialog")
+            case .engagementAccepted:
+                switch coordinator.presentation {
+                case let .embedded(_, onEngagementAccepted: callback):
+                    callback()
+                case .alert:
+                    break
+                }
+            }
+        }
+
+        coordinator.start()
+
+        self.visitorCodeCoordinator = coordinator
+    }
+
+    func handleAcceptedUpgrade() {
+        guard videoCallCoordinator == nil else { return }
+        showVideoCallViewController()
+    }
+
+    func handleEngagementRequest(
+        request: CoreSdkClient.Request,
+        answer: Command<Bool>
+    ) {
+        guard request.outcome == .timedOut else {
+            handleEngagementRequestOutcomeNil(answer: answer)
+            return
+        }
+        handleEngagementRequestOutcomeTimeout(answer: answer)
+    }
+
+    func handleEngagementRequestOutcomeTimeout(answer: Command<Bool>) {
+        environment.alertManager.dismissCurrentAlert()
+        environment.gcd.mainQueue.asyncAfterDeadline(.now() + 0.5) {
+            // Will be swapped for localized string
```

Comment:
What about this? Is there a ticket to provide localised strings?

Reply:
Ticket for it has been created and comment updated [ad3bf2e](https://github.com/salemove/ios-sdk-widgets/pull/1161/commits/ad3bf2e94dbeec8044b15a5b2a7be28472e02139)